### PR TITLE
sdxl pipeline refactor

### PR DESCRIPTION
**Pull request by ppadjinTT**
### Problem description
Small refactor of SDXL pipeline example for better readability.

### What's changed
- warmup argument in `setup` replaces the need for manual warmup inference run
- `uncond` prompts renamed to `negative`, more consistent with most APIs
- improved type hinting

### CODE REVIEW

File: examples/pytorch/sdxl-pipeline.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 import time
 from enum import Enum
+from typing import Optional
 
 import torch
 import torch_xla
 import torch_xla.core.xla_model as xm
 import torch_xla.runtime as xr
 from diffusers import AutoencoderKL, EulerDiscreteScheduler, UNet2DConditionModel
+from jaxtyping import Float
```
[the remark this answers is not in the document]
Comment (ppadjinTT) [reply examples/pytorch/sdxl-pipeline.py]: weirdly enough this is the only way i found to cleanly display the return tensor shape in function signature `Float[torch.Tensor, "B 3 H W"]` - this was actually useful to note what is the exact return shape
